### ISSUE TITLE
attn_mask for GATConv.forward

### ISSUE THREAD

**Issue by AzOuss96** [rusty1s/pytorch_geometric#2835]
Hello,

Is there a way to mask specific entries like in [torch.MultiheadAttention.forward](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html) using the attn_mask attribute?

Thank you,

**Comment by rusty1s** [comment]
I suggest that you mask out the graph connectivity instead, e.g., via `edge_index[:, mask]`.

**Comment by AzOuss96** [comment]
Thank you! It seems that it does not support a `batch_size` dimension in the forward method, I want to use the GATConv layer with a features tensor `x` of shape `(batch_size, n_nodes, in_channels)` and adjacency edge indices `edge_index` of shape `(batch_size, 2, n_edges)`

**Comment by rusty1s** [comment]
Mini-batching is a little bit different in PyG, in particular, there is no dedicated batch dimension. You can read more about it [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/introduction.html#mini-batches).

**Comment by AzOuss96** [comment]
Thank you!
